nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -221,8 +221,19 @@ in
             Group = "dhcpcd";
             StateDirectory = "dhcpcd";
             RuntimeDirectory = "dhcpcd";
+
+            ExecStartPre = "+${pkgs.writeShellScript "migrate-dhcpcd" ''
+              # migrate from old database directory
+              if test -d /var/db/dhcpcd; then
+                mv /var/db/dhcpcd/* -t /var/lib/dhcpcd
+                rmdir /var/db/dhcpcd
+                chown -R dhcpcd:dhcpcd /var/lib/dhcpcd
```

Comment:
Isn't this running as unprivileged user? Are you sure chmod works?

Reply:
Yes, it works: the trick is the `+` at the beginning. (See "Table 2. Special executable prefixes" in `systemd.service(5)`)